### ISSUE TITLE
load over 350 images with Pimage 

### ISSUE THREAD

**Issue by yanghansyuan** [processing/processing#4112]
Hello,I want to display muti-video with alpha channel so I use sequence file with png. But I found out there's limitation amount of image loading? when ever I want to load over 350 images it gots error:

*java.lang.RuntimeException
	at processing.opengl.PSurfaceJOGL$2.run(PSurfaceJOGL.java:443)
	at java.lang.Thread.run(Thread.java:745)*

here is my simple code:
```
PImage[] images;
int imageCount;

void setup() {
  frameRate(30);
  size(960, 540, P3D);  

  imageCount = 350;
  images = new PImage[imageCount];

  for (int i = 0; i < imageCount; i++) {
    images[i] = loadImage("00000.png");
  }
}


void draw() {
  background(0); 
}
```

Is there any memory limitation or something? Thanks for any help.

**Comment by JakubValtar** [comment]
Hi @yanghansyuan, it's probably a problem with memory. It should throw something like:
```
java.lang.RuntimeException: java.lang.OutOfMemoryError: Java heap space
	at processing.opengl.PSurfaceJOGL$2.run(PSurfaceJOGL.java:443)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.OutOfMemoryError: Java heap space
	at processing.core.PImage.<init>(PImage.java:332)
	at processing.core.PApplet.loadImage(PApplet.java:5327)
	at processing.core.PApplet.loadImage(PApplet.java:5230)
```

If you do the math, 350 frames will take 672 MB for 800x600 frames and 2.9 GB for 1920x1080 frames in your RAM and also in your graphics card memory. For that you need 64bit Java with tweaked VM and beefy graphics card. And that's only 14 seconds.

Better approach would be to have two videos, one with colors and the other just grayscale with alpha. You can mask them in code or in a simple shader (that would be super fast). To play them would require only a fraction of memory and you could have much longer videos. Please ask on [the forum](https://forum.processing.org/two/) for help with the code.

**Comment by yanghansyuan** [comment]
Thank you @JakubValtar 

I did some research, all the ppl suggest the same way as you, use mask(). However I always got some visual sync problem in between colors video and alpha video.(Happened both in video or sequence images.)  So that I'm trying to load sequence PNG images with alpha channel.

From what I have now,as your advice, I increase maximum available memory up to 3.5GB. And for loading sequence images,I use "requestImage" instead of "loadImage" and run pretty ok now. I will keep post what's going on. And I will keep find out how the "mask" approach work too.

now sure is it appropriate to discuss this in here. Here seems to report issue of core processing and what I bump into now is more like a technical issue. If it's not ok I will switch to forum for ask more help.

thank for your help.
